solr/modules/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java

```diff
@@ -233,7 +235,14 @@ protected static boolean scoreSingleHit(
     boolean logHit = false;
 
     scorer.getDocInfo().setOriginalDocScore(hit.score);
+    QueryLimits queryLimits = QueryLimits.getCurrentLimits();
     hit.score = scorer.score();
+    if (queryLimits.maybeExitWithPartialResults(
```

Comment:
This method doesn't throw exception and returns a boolean only when partial results are acceptable - meaning that since we already computed some expensive results we can still decide here to return them if we want to. We can also return with empty results if nothing was computed yet.
However, the method `maybeExitWithPartialResults` will throw an exception if the user decided he doesn't want partial results (by setting request param `partialResults=false`) - so we don't need to explicitly throw any exception here.

Reply:
Hi 
@sigram, I see your point now, I added a test and a couple of refinements with the latest commit, I guess it's ok now?